### PR TITLE
load additional column when loading aggregates

### PR DESCRIPTION
**Pull request by talIguaz**

### CODE REVIEW

File: storey/flow.py [hunk 1]
```diff
@@ -561,6 +561,28 @@ def __init__(self, table_path, storage, partitioned_by_key=True):
     def __getitem__(self, key):
         return self._cache[key]
 
+    def __setitem__(self, key, value):
+        self._cache[key] = value
+
+    def update_key(self, key, data):
+        if key not in self._cache:
```
Comment (gtopper) [inline storey/flow.py]: Maybe without negation?

**Review by gtopper — APPROVED** [review]

File: storey/aggregations.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ class AggregateByKey(Flow):
     :type augmentation_fn: Function ((Event, dict) => Event)
     """
 
-    def __init__(self, aggregates, table, key=None, emit_policy=_default_emit_policy, augmentation_fn=None):
+    def __init__(self, aggregates, table, key=None, emit_policy=_default_emit_policy, augmentation_fn=None, enrich_with=None, aliases=None):
```
Comment (gtopper) [inline storey/aggregations.py:36]: Need to add these to the doc I guess?

File: storey/flow.py [hunk 3]
```diff
@@ -618,6 +618,28 @@ def __init__(self, table_path, storage, partitioned_by_key=True):
...
+        self._cache[key] = value
+
+    def update_key(self, key, data):
+        if key in self._cache:
+            for name, value in data.items():
+                self._cache[key][name] = value
+        else:
+            self._cache[key] = data
+
+    async def lazy_load_key_with_aggregates(self, key, timestamp=None):
```
Comment (gtopper) [inline storey/flow.py:631]: Any reason for this to be async?

Comment (talIguaz) [reply storey/flow.py:631]: well it potentially does an IO

**Review by gtopper — APPROVED** [review]

**Pull request merged**